src/main/java/com/juniorstart/juniorstart/controller/UserProfileController.java

```diff
@@ -0,0 +1,44 @@
+package com.juniorstart.juniorstart.controller;
+
+
+import com.juniorstart.juniorstart.model.UserProfile;
+import com.juniorstart.juniorstart.service.UserProfileService;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/user/profile")
+public class UserProfileController {
+
+    private UserProfileService userProfileService;
+
+    public UserProfileController(UserProfileService userProfileService){
+        this.userProfileService = userProfileService;
+    }
+
+    @PostMapping("")
+    public ResponseEntity<UserProfile> addUserProfile(@RequestBody UserProfile userProfile){
+        userProfileService.addUserProfile(userProfile);
+        return ResponseEntity.ok(userProfile);
+    }
+
+    @PutMapping("")
+    public ResponseEntity<UserProfile> updateUserProfile(@RequestBody UserProfile userProfile){
+        userProfileService.updateUserProfile(userProfile);
+        return ResponseEntity.ok(userProfile);
+    }
+
+    @GetMapping("/technology_role")
+    public ResponseEntity<?>  findMentorByTechnology(@RequestParam List<String> technology, List<String> userRole){
```

Comment:
Lepiej przekazać obiekt z wartościami

Reply:
@hosu794  Masz na myśli listę parametrów jako obiekt? Jeśli tak, to czy ma to jakiś większy sens ? Bo wtedy zamiast Listy/Stringa trzeba przekazać całą encję.